src/main/java/frc/robot/commands/DriveTeleopCommand.java

```diff
@@ -0,0 +1,56 @@
+package frc.robot.commands;
+
+import edu.wpi.first.math.geometry.Rotation2d;
+import edu.wpi.first.math.kinematics.DifferentialDriveOdometry;
+import edu.wpi.first.wpilibj.XboxController;
+import edu.wpi.first.wpilibj.smartdashboard.Field2d;
+import edu.wpi.first.wpilibj.smartdashboard.SmartDashboard;
+import edu.wpi.first.wpilibj2.command.Command;
+import frc.robot.subsystems.DriveSubsystem;
+
+
+public class DriveTeleopCommand extends Command {
+    private final DriveSubsystem driveSubsystem;
+    private final XboxController xboxController;
+    private final DifferentialDriveOdometry odometry;
+    private final Field2d field2d;
+
+
+    public DriveTeleopCommand(DriveSubsystem driveSubsystem, XboxController xboxController) {
+        this.driveSubsystem = driveSubsystem;
+        this.xboxController = xboxController;
+        this.field2d = new Field2d();
+        addRequirements(this.driveSubsystem);
+        this.odometry = new DifferentialDriveOdometry(new Rotation2d(this.driveSubsystem.getAngleDegrees()), this.driveSubsystem.getLeftDistancePassedMeters(), this.driveSubsystem.getRightDistancePassedMeters());
```

Comment:
no no no, you misunderstood using odometry. 

The point is to track the position of the robot on the field from the start to stop. For this to work, we must track and update odometery _always_. And the use for it is less teleop and more autonomous. 

As such, this should not be in a command, but rather in the subsystem.